### PR TITLE
Add Echo parameter to llama runner, jni+java layer, and demo app

### PR DESCRIPTION
**Pull request by cmodi-meta**
To allow the developer to set whether the response should echo (i.e. include) the input prompt.
 - echo = true then token_callback for output will also include input (i.e. Text Completion).
 - echo = false then token_callback for output will not include input (i.e. Chat generation).

We'll default to true since that will maintain the existing flow.

### CODE REVIEW

**Review by kirklandsign — APPROVED** [review]

**Comment by kirklandsign** [comment]
linter 😶‍🌫️

**Review by larryliu0820 — APPROVED** [review]
Looks good!

File: examples/models/llama2/runner/runner.cpp [hunk 1]
```diff
@@ -204,16 +205,20 @@ Error Runner::generate(
...
   stats_.first_token_ms = util::time_in_ms();
   stats_.prompt_eval_end_ms = util::time_in_ms();
   ET_CHECK_OK_OR_RETURN_ERROR(prefill_res.error());
   uint64_t cur_token = prefill_res.get();
 
   // print the first token from prefill. No prev_token so use cur_token for it.
-  wrapped_callback(ET_UNWRAP(tokenizer_->decode(cur_token, cur_token)));
+  if (echo) {
+    wrapped_callback(ET_UNWRAP(tokenizer_->decode(cur_token, cur_token)));
+  }
```
Comment (larryliu0820) [inline examples/models/llama2/runner/runner.cpp:221]: I think this is needed, otherwise we are missing the first generated token.

Comment (cmodi-meta) [reply examples/models/llama2/runner/runner.cpp:221]: from my testing, this callback would just return a newline. We also tested this and the output works well.

Comment (larryliu0820) [reply examples/models/llama2/runner/runner.cpp:221]: Hmm I'm saying it doesn't matter what that token is. Semantically it is the first token the model generates, by prefilling the prompt, so we should trigger callback on this one.

Comment (cmodi-meta) [reply examples/models/llama2/runner/runner.cpp]: got it! @larryliu0820, I've removed the echo around the first token generation.

File: examples/demo-apps/android/LlamaDemo/app/src/main/java/com/example/executorchllamademo/MainActivity.java [hunk 2]
```diff
@@ -73,8 +73,15 @@ public class MainActivity extends AppCompatActivity implements Runnable, LlamaCa
 
   @Override
   public void onResult(String result) {
-    mResultMessage.appendText(result);
-    run();
+    if(result.equals("\n\n")) {
```
Comment (cmodi-meta) [inline examples/demo-apps/android/LlamaDemo/app/src/main/java/com/example/executorchllamademo/MainActivity.java:76]: @Riandy fyi with this check on the app side.

**Review by larryliu0820 — APPROVED** [review]

**Pull request merged**